src/Kentico.Xperience.Ecommerce.Common/Admin/FormComponentConfigurator/ContentFolderWorkspaceConfigurator.cs

```diff
@@ -0,0 +1,46 @@
+﻿using CMS.ContentEngine;
+using CMS.Membership;
+
+using Kentico.Xperience.Admin.Base.Forms;
+using Kentico.Xperience.Ecommerce.Common.Admin.FormComponentConfigurator;
+
+[assembly: RegisterFormComponentConfigurator(ContentFolderWorkspaceConfigurator.IDENTIFIER, typeof(ContentFolderWorkspaceConfigurator))]
+
+namespace Kentico.Xperience.Ecommerce.Common.Admin.FormComponentConfigurator;
+
+public class ContentFolderWorkspaceConfigurator : FormComponentConfigurator<ContentFolderSelectorComponent>
+{
+    public const string IDENTIFIER = "ContentFolderWorkspaceConfigurator";
+
+    private readonly IContentFolderManager contentFolderManager;
+
+    public ContentFolderWorkspaceConfigurator(IContentFolderManagerFactory contentFolderManagerFactory)
+        => contentFolderManager = contentFolderManagerFactory.Create(UserInfoProvider.AdministratorUser.UserID);
+
+    public override Task Configure(ContentFolderSelectorComponent formComponent, IFormFieldValueProvider formFieldValueProvider, CancellationToken cancellationToken)
+    {
+        if (!formFieldValueProvider.TryGet(nameof(SynchronizationSettingsModel.WorkspaceName), out string? workspaceName) || string.IsNullOrEmpty(workspaceName))
+        {
+            formComponent.AddVisibilityCondition(new Invisible());
+            return Task.CompletedTask;
+        }
+
+        formComponent.Properties.WorkspaceName = workspaceName;
+        return Task.CompletedTask;
+    }
+
+    public override async Task<object> ConfigureValue(IFormFieldValueProvider formFieldValueProvider, string changedFieldName, CancellationToken cancellationToken)
+    {
+        if (changedFieldName == nameof(SynchronizationSettingsModel.WorkspaceName) && formFieldValueProvider.TryGet(nameof(SynchronizationSettingsModel.WorkspaceName), out string? workspaceName) && !string.IsNullOrEmpty(workspaceName))

```

Comment:
We should use `Equals()` method for comparing strings. The `==` operator has it's advantages (by default it evaluates `null` comparison without throwing an exception, however, having an explicit null-check in this case seems to be more fitting.